ast/Trees.h

```diff
@@ -235,11 +239,16 @@ struct ParsedFile {
 class ParsedFilesOrCancelled final {
 private:
     std::optional<std::vector<ParsedFile>> trees;
+    ParsedFilesOrCancelled();
```

Comment:
I made this private to enforce that callsites must destroy trees in parallel, in the hopes of preventing a regression in the future.